scipy/spatial/distance.py

```diff
@@ -144,6 +146,88 @@ def _validate_vector(u, dtype=None):
         raise ValueError("Input vector should be 1-D.")
     return u
 
+def directed_hausdorff(u, v, seed=0):
+    """
+    Computes the directed Hausdorff distance between two N-D arrays.
+
+    Distances between pairs are calculated using a Euclidean metric.
+    
+    Parameters
+    ----------
+    u : (M,N) ndarray
+        Input array.
+    v : (O,N) ndarray
+        Input array.
+    seed : int or None
+        Local `RandomState` seed. Deafult is 0, a random shuffling of
+        u and v that guarantees reproducibility.
+
+    Returns
+    -------
+    out : tuple
+        The directed Hausdorff distance between arrays `u` and `v`,
+        followed by indices of the Hausdorff pair in `u` and `v`,
+        respectively.
+
+    Notes
+    ----------
+    Uses the early break technique and the random sampling approach
+    described by [1]_. Although worst-case performance is O(m * o)
+    (as with the brute force algorithm), this is unlikely in practice
+    as the input data would have to require the algorithm to explore
+    every single point interaction, and after the algorithm shuffles
+    the input points at that. The best case performance is O(m), which
+    is satisfied by selecting an inner loop distance that is less than
+    cmax and leads to an early break as often as possible. The authors
+    have formally shown that the average runtime is closer to O(m).
+
+    .. versionadded:: 0.19.0
+
+    References
+    ----------
+
```

Comment:
style nitpick: remove blank line, and reindent reference to line up with ``A. A.``